### ISSUE TITLE
Don't use deprecated APIs

### ISSUE THREAD

**Issue by kseo** [denibertovic/docker-hs#24]
Building docker-hs emits -Wdeprecations warnings.

```
/Users/kseo/HaskellProjects/docker-hs/src/Docker/Client/Http.hs:88:26: warning: [-Wdeprecations]
    In the use of ‘parseUrl’
    (imported from Network.HTTP.Client, but defined in http-client-0.4.31.2:Network.HTTP.Client.Request):
    Deprecated: "Please use parseUrlThrow, parseRequest, or parseRequest_ instead"

/Users/kseo/HaskellProjects/docker-hs/src/Docker/Client/Http.hs:130:23: warning: [-Wdeprecations]
    In the use of ‘sClose’ (imported from Network.Socket):
    Deprecated: "use 'close'"
```

If there is no reason to use these deprecated APIs, I will create a PR to replace them.

**Comment by denibertovic** [comment]
@kseo I don't see a reason why we can't switch to the new functions. :+1:  cc @jprider63

**Comment by jprider63** [comment]
👍

**Comment by kseo** [comment]
I created #25

**Comment by denibertovic** [comment]
Closing as this has been merged in #30
